routes/markerSites.js

```diff
@@ -23,15 +23,42 @@ module.exports = function(connection) {
     /* GET all the markers */
     router.get('/markers', function(req, res, next) {
 
-        var query="SELECT * FROM atlas";
+        var query = "SELECT * FROM atlas WHERE";
+
+        //filter by url
+        var criteria = [
+            {
+                param: 'username',
+                col: 'created_by',
+            },
+            {
+                param: 'type',
+                col: 'type',
+            },
+            {
+                param: 'versions',
+                col: 'openmrs_version',
+            },
+            {
+                param: 'dists',
+                col: 'distribution',
+            },
+        ]
+
+        //if parameter exists in the url
+        //push it into params, to use in the sql query
+        var params = [];
+        criteria.forEach(function(crit) {
+            if(req.query[crit.param]) {
+                query += " "+crit.col+"=? AND";
+                params.push(req.query[crit.param]);
+            }
+        });
 
-        //filter by username
-        var username=req.query['username'];
-        if(username) {
-            query += " WHERE created_by='"+username+"'";
-        }
+        if(params.length>0) query=query.substring(0,query.length-3); //If we are filtering, cut out the last 'AND'
+        else query=query.substring(0,query.length-5);   //else, cut out the 'WHERE'
```

Comment:
You can eliminate this hack if you conditionally prepend the ANDs when adding criteria on L53. That is:
`if (params.length > 0) query += whereClause;`

Reply:
That is so much cleaner! I pushed the change. :)